### PR TITLE
Feature/external snapshotter for EKS-A on vSphere and Tester Process for Snow Snapshotting.

### PR DESCRIPTION
**Pull request by elamaran11**
*Issue #, if available:*

*Description of changes:*
 This PR includes installation process for vSphere and Snow Snapshot CRDs, Snapshot Controller, Snapshot classes.
 Also includes tests for Snapshot create and restore validations.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: eks-anywhere-snow/Addons/Core/storage-driver/storage-driver.yaml [hunk 1]
```diff
@@ -1,10 +1,24 @@
+# apiVersion: kustomize.toolkit.fluxcd.io/v1beta2
```
Comment (shapirov103) [inline eks-anywhere-snow/Addons/Core/storage-driver/storage-driver.yaml:1]: why do you need the commented out section? let's drop it.

**Review by shapirov103 — COMMENTED** [review]
a couple of small comment. provided you tested it out looks good overall.

File: eks-anywhere-snow/Addons/Core/storage-driver/storage-driver.yaml (hunk 1 above)
Comment (elamaran11) [reply eks-anywhere-snow/Addons/Core/storage-driver/storage-driver.yaml]: Removed the commented sec.

File: eks-anywhere-snow/Addons/Core/storage-driver/kustomization.yaml [hunk 2]
```diff
@@ -1,4 +1,4 @@
 apiVersion: kustomize.config.k8s.io/v1beta1
 kind: Kustomization
 resources:
-  - storage-driver.yaml
\ No newline at end of file
+  - storage-driver1.yaml
```
[the remark this answers is not in the document]
Comment (elamaran11) [reply eks-anywhere-snow/Addons/Core/storage-driver/kustomization.yaml]: I kept it temp to cleanup. Fixed it now.

**Review by shapirov103 — APPROVED** [review]
LGTM

**Pull request merged**